File: src/src_core/fmg/core/types/draw/FontInfo.java
package fmg.core.types.draw;

import java.beans.PropertyChangeListener;

import fmg.common.geom.DoubleExt;
import fmg.common.notyfier.INotifyPropertyChanged;
import fmg.common.notyfier.NotifyPropertyChanged;

/** minimal font descripton */
public class FontInfo implements INotifyPropertyChanged {

   /** font name */
   private String _name = "SansSerif"; // Arial
   /** font is bold? */
   private boolean _bold = false;
   /** font size */
   private double _size = 10;
   protected NotifyPropertyChanged _notifier = new NotifyPropertyChanged(this);

   public FontInfo() { }
   public FontInfo(String fontName, boolean isBold, int fontSize) {
      _name = fontName;
      _bold = isBold;
      _size = fontSize;
   }

   public static final String PROPERTY_NAME = "Name";
   public static final String PROPERTY_BOLD = "Bold";
   public static final String PROPERTY_SIZE = "Size";

   public String getName() { return _name; }
   public void setName(String fontName) {
      if (_name.equals(fontName))
         return;
      String old = _name;
      _name = fontName;
      _notifier.onPropertyChanged(old, fontName, PROPERTY_NAME);
   }

   public boolean isBold() { return _bold; }
   public void setBold(boolean isBold) {
      if (_bold == isBold)
         return;
      boolean old = _bold;
      _bold = isBold;
      _notifier.onPropertyChanged(old, isBold, PROPERTY_BOLD);
   }

   public double getSize() { return _size; }
   public void setSize(double size) {
      double old = _size;
      if (DoubleExt.hasMinDiff(_size, size))
         return;
      _size = size;
      _notifier.onPropertyChanged(old, size, PROPERTY_SIZE);
   }

   @Override
   public int hashCode() {
      int result = 31 + ((_name == null) ? 0 : _name.hashCode());
      result = 31 * result + (_bold ? 1231 : 1237);
      long temp = Double.doubleToLongBits(_size);
      return 31 * result + (int)(temp ^ (temp >>> 32));
   }

   @Override
   public boolean equals(Object obj) {
      if (this == obj)
         return true;
      if (obj == null)
         return false;
      if (getClass() != obj.getClass())
         return false;
      FontInfo other = (FontInfo)obj;
      if (_name == null) {
         if (other._name != null)
            return false;
      }
      return _name.equals(other._name) &&
            (_bold == other._bold) &&
            DoubleExt.hasMinDiff(_size, other._size);
   }

   @Override
   public String toString() {
      return "FontInfo{fontName=" + _name + ", isBold=" + _bold + ", size=" + _size + "}";
   }

   @Override
   public void addListener(PropertyChangeListener listener) {
      _notifier.addListener(listener);
   }
   @Override
   public void removeListener(PropertyChangeListener listener) {
      _notifier.removeListener(listener);
   }
}

File: src/src_core/fmg/core/types/draw/FontInfo.cs
using System;
using System.ComponentModel;
using fmg.common.geom;
using fmg.common.notifier;

namespace fmg.core.types.draw {

    public class FontInfo : INotifyPropertyChanged, IDisposable {

        private string _name = "Arial"; // Times New Roman // Verdana // Courier New // SansSerif;
        private bool _bold = false;
        private double _size = 10;
        public event PropertyChangedEventHandler PropertyChanged;
        protected readonly NotifyPropertyChanged _notifier;

        public FontInfo() {
            _notifier = new NotifyPropertyChanged(this);
            _notifier.PropertyChanged += OnNotifierPropertyChanged;
        }

        public string Name {
            get { return _name; }
            set { _notifier.SetProperty(ref _name, value); }
        }

        public bool Bold {
            get { return _bold; }
            set { _notifier.SetProperty(ref _bold, value); }
        }

        public double Size {
            get { return _size; }
            set {
                //System.Diagnostics.Debug.Assert(value > 0.01);
                if (value < 0.01) {
                    //throw new ArgumentException("Font size value must be positive: value=" + value);
                    System.Diagnostics.Debug.WriteLine("Font size value must be positive: value=" + value);
                    value = 0.1;
                }
                // _notifier.SetProperty(ref _size, value);
                double old = _size;
                if (_size.HasMinDiff(value))
                    return;
                _size = value;
                _notifier.FirePropertyChanged(old, value);
            }
        }

        protected bool Equals(FontInfo other) {
            return string.Equals(_name, other._name) && (_bold == other._bold) && (_size == other._size);
        }

        public override bool Equals(object obj) {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            return (GetType() == obj.GetType()) && Equals((FontInfo)obj);
        }

        public override int GetHashCode() {
            unchecked {
                var hashCode = _name?.GetHashCode() ?? 0;
                hashCode = (hashCode * 397) ^ _bold.GetHashCode();
                return (hashCode * 397) ^ _size.GetHashCode();
            }
        }

        public override string ToString() {
            return string.Format("FontInfo={{name={0}, bold={1}, size={2}}}", _name, _bold, _size);
        }

        private void OnNotifierPropertyChanged(object sender, PropertyChangedEventArgs ev) {
            System.Diagnostics.Debug.Assert(ReferenceEquals(sender, _notifier));
            PropertyChanged?.Invoke(this, ev);
        }

        public void Dispose() {
            _notifier.PropertyChanged -= OnNotifierPropertyChanged;
            _notifier.Dispose();
            NotifyPropertyChanged.AssertCheckSubscribers(this);
            GC.SuppressFinalize(this);
        }

    }

}
